src/components/pill/mastercard.js

```diff
@@ -28,27 +41,55 @@ const MasterCard = forwardRef(
       ...rest,
     }
 
-    return (
+    const MasterCardComponent = (
       <MasterCardContainer
         background={getMasterCardBackground(
           pillRight.background,
           pillRight.flavour || "disabledWarning"
         )}
-        height={height}
-        onClick={onClick}
-        round={round}
-        size={size}
-        ref={ref}
-        {...(dataTestId && { "data-testid": dataTestId })}
+        data-testid={testId}
+        isAlert={isAlert}
+        {...containerProps}
+        {...(!isAlert && outerContainerProps)}
       >
         {children || (
           <>
-            <MasterCardPill side="left" {...pillProps} {...pillLeft} />
-            <MasterCardPill side="right" {...pillProps} {...pillRight} />
+            <MasterCardPill side="left" isAlert={isAlert} data-testid={`${testId}-left-pill`} {...pillProps} {...pillLeft} />
+            <MasterCardPill side="right" data-testid={`${testId}-right-pill`} {...pillProps} {...pillRight} />
           </>
         )}
       </MasterCardContainer>
     )
+
+    if (!isAlert) return MasterCardComponent
```

Comment:
Why do we have this coupling? `Alert` looks like it is a flavour, hence I would expect `Mastercard` to be plain, and use a separate component `AlertMastercard` to accomodate the specifics for alerts. Can we decouple this please?

Reply:
Sure! 😄